api/v1alpha1/service_types.go

```diff
@@ -45,30 +48,34 @@ type ServiceFilter struct {
 	// +optional
 	Name *OpenStackName `json:"name,omitempty"`
 
-	// description of the existing resource
+	// type of the existing resource
 	// +kubebuilder:validation:MinLength:=1
 	// +kubebuilder:validation:MaxLength:=255
 	// +optional
-	Description *string `json:"description,omitempty"`
-
-	// TODO(scaffolding): Add more types.
-	// To see what is supported, you can take inspiration from the ListOpts stucture from
-	// github.com/gophercloud/gophercloud/v2/openstack/identity/v3/services
+	Type *string `json:"type,omitempty"`
 }
 
 // ServiceResourceStatus represents the observed state of the resource.
 type ServiceResourceStatus struct {
-	// name is a Human-readable name for the resource. Might not be unique.
-	// +kubebuilder:validation:MaxLength=1024
+	// name indicates the name of service.
+	// +kubebuilder:validation:MaxLength:=255
 	// +optional
 	Name string `json:"name,omitempty"`
 
-	// description is a human-readable description for the resource.
-	// +kubebuilder:validation:MaxLength=1024
+	// description indicates the description of service.
+	// +kubebuilder:validation:MinLength:=1
```

Comment:
let's drop the MinLength check in the resource status (we should make that part of the linter if we figure out how to do it).